evalml/pipelines/components/transformers/preprocessing/polynomial_decomposer.py

```diff
@@ -180,20 +185,80 @@ def inverse_transform(self, y_t: pd.Series) -> tuple[pd.DataFrame, pd.Series]:
         Raises:
             ValueError: If y is None.
         """
-        _, y_t = self._check_target(None, y_t)
-
-        y_t = infer_feature_types(y_t)
+        if y_t is None:
+            raise ValueError("y_t cannot be None for Decomposer!")
+        original_index = y_t.index
+
+        y_t = infer_feature_types(y_t).copy()
+        self._check_oos_past(y_t)
+
+        index = self._choose_proper_index(y_t)
+
+        y_in_sample = pd.Series([])
+        y_out_of_sample = pd.Series([])
+
+        # For partially and wholly in-sample data, retrieve stored results.
+        if index[0] <= y_t.index[0] <= index[-1]:
+            left_index = y_t.index[0]
+            right_index = min(y_t.index[-1], index[-1])
+
+            if isinstance(y_t.index, (pd.RangeIndex, Int64Index)):
+                y_t_ind = pd.RangeIndex(
+                    start=left_index,
+                    stop=right_index + 1,
+                )  # stop value is not inclusive
+            elif isinstance(y_t.index, pd.DatetimeIndex):
+                y_t_ind = pd.DatetimeIndex(
+                    pd.date_range(left_index, end=right_index),
+                )  # end value is inclusive
+
+            # Build index
+            y_t_in_sample = y_t[y_t_ind]
+
+            # Convert y_t to datetime index to use the built in component
+            if isinstance(y_t.index, (pd.RangeIndex, Int64Index)):
+                y_t_dt_ind = self._convert_int_index_to_dt_index(y_t_in_sample.index)
+            elif isinstance(y_t.index, pd.DatetimeIndex):
+                y_t_dt_ind = y_t_ind
+            trend = self._component_obj.inverse_transform(
+                y_t_in_sample.set_axis(y_t_dt_ind),
+            )
```

Comment:
This has been the main offender and is one of the main differences between the `STLDecomposer` and `PolynomialDecomposer`.  With respect to the trend, the PD's underlying object has the same fit/transform/inverse_transform API that our components do, so it fits well.  The STLD does not have this so requires a lot of code to do manual trend projection.  I believe that both required the detrended target signal to have datetime indices, which is challenging.

Reply:
just to clarify - this new block ensures that both STL and PD are using the same logic when calculating `inverse_transform` and not depend on PD's native `inverse_transform`?